Framework/Crystal/src/ConvertPeaksWorkspace.cpp

```diff
@@ -0,0 +1,169 @@
+                                                                       PropertyMode::Optional),
+                  "Donor PeaksWorkspace with instrument for conversion");
+
+  // output
+  declareProperty(std::make_unique<WorkspaceProperty<IPeaksWorkspace>>("OutputWorkspace", "", Direction::Output),
+                  "Converted Workspaces");
+}
+
+/**
+ * @brief Inputs validation
+ *
+ * @return std::map<std::string, std::string>
+ */
+std::map<std::string, std::string> ConvertPeaksWorkspace::validateInputs() {
+  std::map<std::string, std::string> issues;
+
+  IPeaksWorkspace_sptr ipws = getProperty("PeakWorkspace");
+  PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(ipws);
+  LeanElasticPeaksWorkspace_sptr lpws = std::dynamic_pointer_cast<LeanElasticPeaksWorkspace>(ipws);
+
+  // case I: missing instrument when converting to PeaksWorkspace
+  if (lpws && !pws) {
+    if (getPointerToProperty("InstrumentWorkpace")->isDefault()) {
+      issues["InstrumentWorkpace"] = "Need a PeaksWorkspace with proper instrument attached to assist conversion.";
+    }
+  }
+
+  return issues;
+}
+
+/**
+ * @brief Algorithm entrance func
+ *
+ */
+void ConvertPeaksWorkspace::exec() {
+  // parsing input
+  IPeaksWorkspace_sptr ipws = getProperty("PeakWorkspace");
+  PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(ipws);
+  LeanElasticPeaksWorkspace_sptr lpws = std::dynamic_pointer_cast<LeanElasticPeaksWorkspace>(ipws);
+
+  // decide which route to take
+  if (pws && !lpws) {
+    g_log.notice() << "PeaksWorkspace -> LeanElasticPeaksWorkspace\n";
+    IPeaksWorkspace_sptr outpws = makeLeanElasticPeaksWorkspace(ipws);
+    setProperty("OutputWorkspace", outpws);
+  } else {
+    g_log.notice() << "LeanElasticPeaksWorkspace -> PeaksWorkspace\n";
+    IPeaksWorkspace_sptr ws = getProperty("InstrumentWorkpace");
+    IPeaksWorkspace_sptr outpws = makePeaksWorkspace(ipws, ws);
+    setProperty("OutputWorkspace", outpws);
+  }
+
+  // cleanup
+}
+
+/**
+ * @brief make a LeanElasticPeaksWorkspace using peaks from a regular PeaksWorkspace
+ *
+ * @param ipws input PeaksWorkspace as IPeaksWorkspace_sptr
+ * @return IPeaksWorkspace_sptr
+ */
+IPeaksWorkspace_sptr ConvertPeaksWorkspace::makeLeanElasticPeaksWorkspace(IPeaksWorkspace_sptr ipws) {
+  // prep
+  PeaksWorkspace_sptr pws = std::dynamic_pointer_cast<PeaksWorkspace>(ipws);
+  LeanElasticPeaksWorkspace_sptr lpws = std::make_shared<LeanElasticPeaksWorkspace>();
+
+  ExperimentInfo_sptr inputExperimentInfo = std::dynamic_pointer_cast<ExperimentInfo>(ipws);
+  lpws->copyExperimentInfoFrom(inputExperimentInfo.get());
+
+  // down casting Peaks to LeanElasticPeaks
+  for (int i = 0; i < pws->getNumberPeaks(); ++i) {
+    LeanElasticPeak lpk(pws->getPeak(i));
+    // NOTE:
+    // Peak level info requires explicit copying
+    lpk.setRunNumber(pws->getPeak(i).getRunNumber());
```

Comment:
I don't think these 3 lines should be required, They are all part of the BasePeak so should be copied over during construction, if not then there is a problem in LeanElasticPeak. You can add these to the test to check that it does the correct tihng.

Reply:
Unfortunately they are required here since the constructor `LeanElasticPeaks(Peak)` is only setting QSample and the Wavelength while ignoring the rest fields.

Reply:
Turns out I was wrong (:sweat: not sure why it wasn't working before).
There is no need to add the info add the Peak level as the base class constructor does take care of all the relevant fields.